src/ModelContextProtocol/Diagnostics.cs

```diff
@@ -34,4 +37,79 @@ internal static Histogram<double> CreateDurationHistogram(string name, string de
         HistogramBucketBoundaries = [0.01, 0.02, 0.05, 0.1, 0.2, 0.5, 1, 2, 5, 10, 30, 60, 120, 300],
     };
 #endif
+
+    internal static ActivityContext ExtractActivityContext(this DistributedContextPropagator propagator, IJsonRpcMessage message)
+    {
+        string? traceparent = null;
+        string? tracestate = null;
+        propagator?.ExtractTraceIdAndState(message, ExtractContext, out traceparent, out tracestate);
+        ActivityContext.TryParse(traceparent, tracestate, true, out var activityContext);
+        return activityContext;
+    }
+
+    private static void ExtractContext(object? message, string fieldName, out string? fieldValue, out IEnumerable<string>? fieldValues)
+    {
+        fieldValues = null;
```

Comment:
Why does this need the fieldValues parameter if it's always just null?

Reply:
it's there to support HTTP headers that allow multiple values with the same header name. It's not the case with JSON, we never have multiple values for the same key.

Reply:
Does that mean this code will be changing to handle something with http? Because right now the parameter is only ever set to null

Reply:
ah, no, this method needs to comply with the runtime delegate `PropagatorGetterCallback` signature https://github.com/dotnet/runtime/blob/36f59e838d4b58d4347552efac9842e306ef2353/src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/DistributedContextPropagator.cs#L27

This code will never deal with HTTP.